### ISSUE TITLE
SC1090 does not recognize `~` as a single expansion

### ISSUE THREAD

**Issue by sparr** [koalaman/shellcheck#3022]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1090
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0 and online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
. "${HOME}"/foo.sh
. ~/foo.sh
```

#### Here's what shellcheck currently says:
```
[Line 2:](javascript:setPosition(2, 3))
. "${HOME}"/foo.sh
  ^-- [SC1091](https://www.shellcheck.net/wiki/SC1091) (info): Not following: ./foo.sh was not specified as input (see shellcheck -x).
 
[Line 3:](javascript:setPosition(3, 3))
. ~/foo.sh
  ^-- [SC1090](https://www.shellcheck.net/wiki/SC1090) (warning): ShellCheck can't follow non-constant source. Use a directive to specify location.
```

#### Here's what I wanted or expected to see:
Line 3 should recognize `~` as [approximately] equivalent to `$HOME` and pass SC1090 with the "strip a single expansion followed by a slash" exception described on the wiki, instead behaving like Line 2 and triggering SC1091 (or, if run with `shellcheck -x`, succeeding as Line 2 would)
